backend/src/graphql/resolver/TransactionResolver.ts

```diff
@@ -210,6 +215,289 @@ async function listTransactions(
+    balance.userId = user.id
+    balance.amount = centAmount
+    balance.modified = received
+  } else {
+    const decaiedBalance = await calculateDecay(balance.amount, balance.recordDate, received).catch(
+      () => {
+        throw new Error('error by calculating decay')
+      },
+    )
+    balance.amount = Number(decaiedBalance) + centAmount
+    balance.modified = new Date()
+  }
+  if (balance.amount <= 0) {
+    throw new Error('error new balance <= 0')
+  }
+  balance.recordDate = received
+  return queryRunner.manager.save(balance).catch((error) => {
+    throw new Error('error saving balance:' + error)
+  })
+}
+
+// helper helper function
+async function addUserTransaction(
+  user: dbUser,
+  transaction: DbTransaction,
+  centAmount: number,
+  queryRunner: QueryRunner,
+): Promise<DbUserTransaction> {
+  let newBalance = centAmount
+  const userTransactionRepository = getCustomRepository(UserTransactionRepository)
+  const lastUserTransaction = await userTransactionRepository.findLastForUser(user.id)
+  if (lastUserTransaction) {
+    newBalance += Number(
+      await calculateDecay(
+        Number(lastUserTransaction.balance),
+        lastUserTransaction.balanceDate,
+        transaction.received,
+      ).catch(() => {
+        throw new Error('error by calculating decay')
+      }),
+    )
+  }
+
+  if (newBalance <= 0) {
+    throw new Error('error new balance <= 0')
+  }
+
+  const newUserTransaction = new DbUserTransaction()
+  newUserTransaction.userId = user.id
+  newUserTransaction.transactionId = transaction.id
+  newUserTransaction.transactionTypeId = transaction.transactionTypeId
+  newUserTransaction.balance = newBalance
+  newUserTransaction.balanceDate = transaction.received
+
+  return queryRunner.manager.save(newUserTransaction).catch((error) => {
+    throw new Error('Error saving user transaction: ' + error)
+  })
+}
+
+// helper function
+/**
+ *
+ * @param senderPublicKey as hex string
+ * @param recipiantPublicKey as hex string
+ * @param amount as float
+ * @param memo
+ * @param groupId
+ */
+async function sendCoins(
+  senderUser: dbUser,
+  recipiantPublicKey: string,
+  amount: number,
+  memo: string,
+  groupId = 0,
+): Promise<boolean> {
+  if (senderUser.pubkey.length !== 32) {
+    throw new Error('invalid sender public key')
+  }
+  if (!isHexPublicKey(recipiantPublicKey)) {
+    throw new Error('invalid recipiant public key')
+  }
+  if (amount <= 0) {
+    throw new Error('invalid amount')
+  }
+  if (!hasUserAmount(senderUser, amount)) {
+    throw new Error("user hasn't enough GDD")
+  }
+
+  const userRepository = getCustomRepository(UserRepository)
+  const recipiantUser = await userRepository.findByPubkeyHex(recipiantPublicKey)
+  if (recipiantUser && recipiantUser.disabled) {
+    throw new Error('recipiant user account is disabled')
+  }
+
+  const centAmount = Math.trunc(amount * 10000)
+
+  // no group id is given so we assume it is a local transfer
+  if (!groupId) {
```

Comment:
this has no `else` block ?! The function will `return true` without doing anything  `if groupId`